internal/pkg/earpointer/heap.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package earpointer
+
+import (
+	"go/types"
+	"strings"
+
+	"golang.org/x/tools/go/ssa"
+)
+
+// An abstract reference is a local (register) or global variable, possibly
+// annotated with some kind of context (call stack, allocation site etc).
+
+// Reference is the base class for local and global references.
+type Reference interface {
+	Type() types.Type // Return the data type of the underlying local or global
+	String() string
+	Node() ssa.Value
+}
+
+// ReferenceContext represents the calling context of a reference.
+// Typically, it contains a stack of call instructions.
+type ReferenceContext []ssa.Instruction
```

Comment:
Is there another *Context that appears in this package? If not, is earpointer.Context sufficiently clear for a user of this package?

(This question applies to ReferenceLocal, Global, )